packages/react-components/react-positioning/src/usePositioningOptions.ts

```diff
@@ -0,0 +1,179 @@
+import type { Middleware } from '@floating-ui/dom';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import * as React from 'react';
+
+import {
+  shift as shiftMiddleware,
+  flip as flipMiddleware,
+  coverTarget as coverTargetMiddleware,
+  maxSize as maxSizeMiddleware,
+  resetMaxSize as resetMaxSizeMiddleware,
+  offset as offsetMiddleware,
+  intersecting as intersectingMiddleware,
+  matchTargetSize as matchTargetSizeMiddleware,
+} from './middleware';
+import type { PositioningConfigurationFn, PositioningConfigurationFnOptions, PositioningOptions } from './types';
+import { toFloatingUIPlacement, hasScrollParent, normalizeAutoSize } from './utils';
+import { devtoolsCallback } from './utils/devtools';
+import { usePositioningConfiguration } from './PositioningConfigurationContext';
+
+/**
+ * @internal
+ *
+ * This is redundant and exists only to manage React dependencies properly & avoid leaking individual options to the
+ * scope of `usePositioningOptions`.
+ */
+function usePositioningConfigFn(
+  configFn: PositioningConfigurationFn,
+  options: PositioningOptions,
+): (container: HTMLElement, arrow: HTMLElement | null) => PositioningConfigurationFnOptions {
+  const {
+    align,
+    arrowPadding,
+    autoSize,
+    coverTarget,
+    flipBoundary,
+    offset,
+    overflowBoundary,
+    pinned,
+    position,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    unstable_disableTether,
+    strategy,
+    overflowBoundaryPadding,
+    fallbackPositions,
+    useTransform,
+    matchTargetSize,
+    shiftToCoverTarget,
+  } = options;
+
+  return React.useCallback(
+    (container: HTMLElement, arrow: HTMLElement | null) => {
+      return configFn({
+        container,
+        arrow,
+        options: {
+          autoSize,
+          matchTargetSize,
+          offset,
+          strategy,
+          coverTarget,
+          flipBoundary,
+          overflowBoundary,
+          useTransform,
+          overflowBoundaryPadding,
+          pinned,
+          arrowPadding,
+          align,
+          fallbackPositions,
+          shiftToCoverTarget,
+          position,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          unstable_disableTether,
+        },
+      });
+    },
+    [
+      autoSize,
+      matchTargetSize,
+      offset,
+      strategy,
+      coverTarget,
+      flipBoundary,
+      overflowBoundary,
+      useTransform,
+      overflowBoundaryPadding,
+      pinned,
+      arrowPadding,
+      align,
+      fallbackPositions,
+      shiftToCoverTarget,
+      position,
+      unstable_disableTether,
+      configFn,
+    ],
+  );
+}
+
+/**
+ * @internal
+ */
+export function usePositioningOptions(options: PositioningOptions) {
+  const { dir, targetDocument } = useFluent();
+  const isRtl = dir === 'rtl';
+
+  const configFn = usePositioningConfigFn(usePositioningConfiguration(), options);
```

Comment:
is the use of the new config fn the only change in this file?

Asking cause moving it out makes the whole file a diff

Reply:
> is the use of the new config fn the only change in this file?

Yes, `usePositioningOptions()` was moved from `usePositioning.ts`, `usePositioningConfigFn()` is a new thing.